File: nim/resistor-color-trio/resistor_color_trio.nim
import math

type
  ResistorColor* = enum
    Black, Brown, Red, Orange, Yellow, Green, Blue, Violet, Grey, White

proc value(colors: openArray[ResistorColor]): int =
  (colors[0].ord * 10 + colors[1].ord) * (10 ^ colors[2].ord)

proc label*(colors: openArray[ResistorColor]): (int, string) =
  let val = value(colors)
  if val >= 1000:
    (val div 1000, "kiloohms")
  else:
    (val, "ohms")

File: csharp/resistor-color-trio/ResistorColorTrio.cs
using System;

public static class ResistorColorTrio
{
    public enum ResistorColor
    {
        Black, Brown, Red, Orange, Yellow, Green, Blue, Violet, Grey, White
    }

    public static string Label(string[] colors)
    {
        var value = Value(colors);
        return value >= 1000 ? $"{value / 1000} kiloohms" : $"{value} ohms";;
    }

    private static int Value(string[] colors) => (Value(colors[0]) * 10 + Value(colors[1])) * (int)Math.Pow(10, Value(colors[2]));

    private static int Value(string color) => (int)Enum.Parse<ResistorColor>(color, ignoreCase: true);
}
